### PR TITLE
upgrade react to v15.5 and use prop-types instead of React.PropTypes

### LINKED ISSUE #1676: Accessing PropTypes via the main React package is deprecated
On updating to the latest version of react, you get this deprecation warning. 
**The fix**: Use the prop-types package from npm instead.

### PR DESCRIPTION
**Pull request by chentsulin**
Use `react-codemod` to transform the prop-types

close #1672, close #1675, close #1676, close #1680

### CODE REVIEW

**Comment by ooade** [comment]
Would have been my first PR here :smile:

**Review by ooade — CHANGES_REQUESTED** [review]
https://github.com/zeit/next.js/blob/master/server/document.js ???

**Comment by arunoda** [comment]
In this case I'd like to see the changes in all of these: https://github.com/zeit/next.js/search?utf8=%E2%9C%93&q=PropTypes&type=

File: package.json [hunk 1]
```diff
@@ -116,14 +117,14 @@
...
-    "react-dom": "15.4.2",
+    "react": "15.5.3",
+    "react-dom": "15.5.3",
     "standard": "9.0.2",
     "wd": "1.2.0"
   },
   "peerDependencies": {
-    "react": "^15.4.2",
-    "react-dom": "^15.4.2"
+    "react": "^15.5.3",
```
Comment (arunoda) [inline package.json]: We don't need to bump this.
Just using `prop-types` is enough.

**Comment by chentsulin** [comment]
Already updated the pr against the comments

**Comment by arunoda** [comment]
Okay. This looks great to me.

**Comment by ooade** [comment]
@bbnnt It's not from your edge. BTW, It's bin resolved but not yet published.

**Comment by bbnnt** [comment]
ok got it, thx for your rep

**Pull request merged**